src/main/java/com/coveo/spillway/exception/SpillwayLimitsWithSameNameException.java

```diff
@@ -1,6 +1,6 @@
 package com.coveo.spillway.exception;
 
-public class SpillwayLimitsWithSameNameException extends SpillwayException {
+public class SpillwayLimitsWithSameNameException extends IllegalArgumentException {
```

Comment:
I think you should leave this as it was, checked, but have the method throw a different RuntimeException.

That way (hopefully) any users will get compiler errors for an exception caught but not thrown, so they at least won't get caught by surprise.

It means the change won't be seamless but I think its preferable to force them to be aware?

Reply:
well I thought about it, and this way, anyone that was handling it, will still be handling it, so I don't think anybody will be caught by surprise

Reply:
oh yeah!